spec/post_show_spec.rb

```diff
@@ -0,0 +1,40 @@
+require 'rails_helper'
+
+RSpec.describe 'Post show page', type: :feature do
+  let!(:user) { User.create(name: 'Sarkodie', photo: '/path/to/photo1.jpg', post_counter: 10) }
+  let!(:post) do
+    Post.create(
+      title: 'Test Post',
+      text: 'This is a test post.',
+      author_id: user.id,
+      comments_counter: 0,
+      likes_counter: 7
+    )
+  end
+
+  describe 'GET #show' do
+    before { visit "/users/#{user.id}/posts/#{post.id}" }
+
+    it 'displays the post title' do
+      expect(page).to have_content(post.title)
+    end
+
+    it 'displays the number of comments' do
+      expect(page).to have_content("Comments: #{post.comments_counter}")
+    end
+
+    it 'displays the number of likes' do
+      expect(page).to have_content("Likes: #{post.likes_counter}")
+    end
+
+    it 'displays the post body' do
+      expect(page).to have_content(post.text)
+    end
+
+    it 'displays the username of each commenter and their comment' do
+      post.comment_entries.each do |comment|
+        expect(page).to have_content("#{comment.user.name} - #{comment.text}")
+      end
+    end
+  end
```

Comment:
- [ ] You are missing the tests for who wrote the post and each comment the commenter left